src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php

```diff
@@ -212,13 +279,22 @@ public function withCount($relations)
                 [$name, $alias] = [$segments[0], $segments[2]];
             }
 
+            // Determine column to do the aggregate function
+            $segments = explode(':', $name);
```

Comment:
Need explanation for these 4 lines or so... no idea what the purpose is and comment is not clear at all.

Reply:
Sorry, I thought the code was self-explanatory. The next lines give us an idea of what they do:

```
if (count($segments) === 2) {
    [$name, $aggregateColumn] = $segments;
}
```

The purpose is to extract the relation name and the column name. For example, calling `User::withSum('posts:earnings')` would mean that we want to retrieve users with a property called `post_earnings_sum` (that's the default name, it can be changed with `AS something`) whose value is the `earnings` column sum of the `posts` relation